packages/desktop-client/src/components/reports/graphs/SpendingGraph.tsx

```diff
@@ -129,7 +137,20 @@ export function SpendingGraph({
   const balanceTypeOp = 'cumulative';
   const thisMonth = monthUtils.currentMonth();
   const lastMonth = monthUtils.subMonths(monthUtils.currentMonth(), 1);
-  const selection = mode.toLowerCase() === 'average' ? 'average' : lastMonth;
+  const lastYear = monthUtils.prevYear(monthUtils.currentMonth());
+  let selection;
+  switch (mode) {
```

Comment:
Your buttons aren't working anymore. You've changed this switch to camelCase but the setMode from where the buttons get clicked don't match.